articles/machine-learning-libraries-in-c++/index.md

```diff
@@ -0,0 +1,191 @@
+6. ```make``` 
+
+#### HOW TO COMPILE PROGRAMS
+ To be able to compile programs with shark, you need to include the header files needed for your use case in machine learning. Let's say for linear regression, you would need to include the following extra header files.
+```C++
+#include <shark/ObjectiveFunction/Loss/Squaredloss.h>
+#include <shark/Algorithms/Trainers/LinearRegression.h>
+```
+You need to link with the following libraries in order to compile.
+```
+-std=c++11 -lboost_serialization-lshark-lcblas
+``` 
+#### IMPLEMENT LINEAR REGRESSION
+ - **Initialize the data**
+ 
+  Include the header files and libraries for linear regression.
+```c++
+#include <bits/stdc++.h> //header file for all basic c++ libraries
+#include <shark/Data/Csv.h> //header file for importing data in csv format
+#include <shark/ObjectiveFunctions/Loss/SquaredLoss.h> //to implement squared loss function
+#include <shark/Algorithms/Trainers/LinearRegression.h>
+```
+Now we need a data set. I have included two .csv files. The independent.csv file includes the x values and the dependent.csv file includes the y values. This is what should be in your mind. ![demo](url/dataset.png). 
+
+The two .csv files can be found on my github [KanizoRGB](https://github.com/KanizoRGB/Machine-learning-libraries.git). Next, create a container to hold the data from the csv files.
+```c++
+Data<RealVector> independent; //store the independent values
+Data<RealVector> dependent;//store the dependent values
+```
+We now need to import the data into our containers. Shark has an `importCSV` function. It is used in the following format
+ ```
+importCSV(datacontainer, "filelocation")
+```
+For our case it thus will be
+```c++
+importCSV(independent, "independent.csv"); // storing the values in specific container by specifying the path of csv
+importCSV(dependent, "dependent.csv");
+```
+Instantiate a regression dataset type and pass our independent and dependent in the constructor for the data. Thereafter, train the linear regression model by instantiating a trainer and define a linear model:
+```c++
+RegressionDataset data(independent, dependent);
+LinearRegression trainer;// trainer for linear regression model
+LinearModel<> model; // linear model
+``` 
+- **Train the model**
+
+ This is how we now train the model. The trainer has a member called train. The member trains the model and finds parameters for it.
+  ```c++
+trainer.train(model, data)
+```
+ - **Prediction**
+ 
+  Let's output the model parameters
+```c++
+// show model parameters
+cout << "intercept: " << model.offset() << endl;
+cout << "matrix: " << model.matrix() << endl;
+```
+The offset member function of linear models outputs the intercept of the best fit line. We also output a matrix instead of a multiplier since the model does not necessarily have to be linear. It can be generalized.Minimize the squared loss to get the line of best fit The line of best fit. Luckily, the model allows us to display all that information.
```

Comment:
```suggestion
The offset member function of linear models outputs the intercept of the best fit line. We also output a matrix instead of a multiplier since the model does not necessarily have to be linear. It can be generalized. Minimize the squared loss to get the line of best fit. Luckily, the model allows us to display all that information.